zerver/lib/markdown/__init__.py

```diff
@@ -1369,7 +1369,13 @@ def handleMatch(self, match: Match[str]) -> Optional[Element]:
         time_input_string = match.group("time")
         timestamp = None
         try:
-            timestamp = dateutil.parser.parse(time_input_string, tzinfos=common_timezones)
+            # Check if the time string is of the new more-readable format.
+            if "|UTC" in time_input_string:
+                # Remove and replace the non-standard characters with the ISO ones.
+                time_format = time_input_string.replace("|", "T", 1).replace("|UTC", "")
```

Comment:
`standard_format_time` might be a better variable name.

Reply:
Done.